src/client/conn/http1.rs

```diff
@@ -88,6 +88,16 @@ where
     pub fn poll_without_shutdown(&mut self, cx: &mut Context<'_>) -> Poll<crate::Result<()>> {
         self.inner.poll_without_shutdown(cx)
     }
+
+    /// Prevent shutdown of the underlying IO object at the end of service the request,
+    /// instead run `into_parts`. This is a convenience wrapper over `poll_without_shutdown`.
+    pub fn without_shutdown(self) -> impl Future<Output = crate::Result<Parts<T>>> {
```

Comment:
I'm thinking that perhaps for 1.0, we make this an `async fn`. It might be a little clearer in the docs, and we can always change from `async fn` to `-> impl Future`, but the other direction I believe is a breaking change.

Reply:
The body of this function, and the one from `server/conn/http1.rs`, is really outdated. I'm not quite sure why it was written like this to begin with (maybe async blocks didn't exist at the time?). Anyways, we should update them:

```rust
pub async fn without_shutdown(mut self) -> crate::Result<Parts<T>> {
    futures_util::future::poll_fn(|cx| self.conn.poll_without_shutdown(cx)).await?;
    Ok(self.into_parts())
}
```
or
```rust
pub fn without_shutdown(mut self) -> impl Future<Output = crate::Result<Parts<T>>> {
    async move {
        futures_util::future::poll_fn(|cx| self.conn.poll_without_shutdown(cx)).await?;
        Ok(self.into_parts())
    }
}
```